t/t1014-mf-priority-dne.t

```diff
@@ -0,0 +1,73 @@
+#!/bin/bash
+
+test_description='Test multi-factor priority plugin with a single user'
+
+. `dirname $0`/sharness.sh
+MULTI_FACTOR_PRIORITY=${FLUX_BUILD_DIR}/src/plugins/.libs/mf_priority.so
+SEND_PAYLOAD=${SHARNESS_TEST_SRCDIR}/scripts/send_payload.py
+SUBMIT_AS=${SHARNESS_TEST_SRCDIR}/scripts/submit_as.py
+DB_PATH=$(pwd)/FluxAccountingTest.db
+
+export TEST_UNDER_FLUX_NO_JOB_EXEC=y
+export TEST_UNDER_FLUX_SCHED_SIMPLE_MODE="limited=1"
+test_under_flux 1 job
+
+flux setattr log-stderr-level 1
+
+test_expect_success 'load multi-factor priority plugin' '
+	flux jobtap load -r .priority-default ${MULTI_FACTOR_PRIORITY}
+'
+
+test_expect_success 'check that mf_priority plugin is loaded' '
+	flux jobtap list | grep mf_priority
+'
+
+test_expect_success 'create flux-accounting DB' '
+	flux account -p $(pwd)/FluxAccountingTest.db create-db
+'
+
+test_expect_success 'add a queue to the DB' '
+	flux account -p ${DB_PATH} add-queue default --priority=0
+'
+
+test_expect_success 'send the queue information to the plugin' '
+	flux account-priority-update -p $(pwd)/FluxAccountingTest.db
+'
+
+test_expect_success 'submit job #1 with an urgency of 0 (will hold the job)' '
+	jobid1=$(flux mini submit --urgency=0 hostname)
+'
+
+test_expect_success 'check that submitted job is in state PRIORITY' '
+	test $(flux jobs -no {state} ${jobid1}) = PRIORITY
+'
+
+test_expect_success 'submit job #2 with a higher priority than job #1' '
+	jobid2=$(flux mini submit --urgency=31 hostname)
+'
+
+test_expect_success 'add some banks to the DB' '
+	flux account -p ${DB_PATH} add-bank root 1 &&
+	flux account -p ${DB_PATH} add-bank --parent-bank=root account1 1
+'
+
+test_expect_success 'add a user to the DB' '
+	username=$(whoami) &&
+	uid=$(id -u) &&
+	flux account -p ${DB_PATH} add-user --username=$username --userid=$uid --bank=account1 --shares=1
+'
+
+test_expect_success 'send the user/bank information to the plugin and reprioritize pending jobs' '
+	flux account-priority-update -p $(pwd)/FluxAccountingTest.db
+'
+
+test_expect_success 'check that job #2 transitions to RUN and job #1 is still pending' '
+	test $(flux jobs -no {state} ${jobid2}) = RUN &&
+	test $(flux jobs -no {state} ${jobid1}) = SCHED
```

Comment:
Testing for the job state _might_ lead to a test race condition. Instead I suggest using `flux job wait-event` and waiting for the `start` event. _If_ the first job is supposed to stay in `SCHED` state, then this test is fine as is, but I'm confused if that should actually be true. (I likely just don't understand the test.)

Reply:
Thanks for pointing this out. I just pushed a change to this test to instead `flux job wait-event $jobid2 start` instead of testing for the job state.